jetbrains-core/src/software/aws/toolkits/jetbrains/services/lambda/sam/sync/SyncApplicationRunProfile.kt

```diff
@@ -0,0 +1,134 @@
+            addParameter("sync")
+            addParameter("--stack-name")
+            addParameter(stackName)
+            addParameter("--template-file")
+            addParameter(templatePath.toString())
+            addParameter("--s3-bucket")
+            addParameter(settings.bucket)
+            settings.ecrRepo?.let {
+                addParameter("--image-repository")
+                addParameter(it)
+            }
+            if (settings.capabilities.isNotEmpty()) {
+                addParameter("--capabilities")
+                addParameters(settings.capabilities.map { it.capability })
+            }
+            if (settings.parameters.isNotEmpty()) {
+                addParameter("--parameter-overrides")
+                settings.parameters.forEach { (key, value) ->
+                    addParameter(
+                        "${escapeParameter(key)}=${escapeParameter(value)}"
+                    )
+                }
+            }
+
+            if (settings.tags.isNotEmpty()) {
+                addParameter("--tags")
+                settings.tags.forEach { (key, value) ->
+                    addParameter(
+                        "${escapeParameter(key)}=${escapeParameter(value)}"
+                    )
+                }
+            }
+            addParameter("--no-dependency-layer")
+            if (syncOnlyCode) {
+                addParameter("--code")
+            }
+        }
+
+        private fun escapeParameter(param: String): String {
+            // Invert the quote if the string is already quoted
+            val quote = if (param.startsWith("\"") || param.endsWith("\"")) {
+                "'"
+            } else {
+                "\""
+            }
+
+            return quote + param + quote
+        }
+
+        override fun execute(executor: Executor, runner: ProgramRunner<*>) = super.execute(executor, runner).apply {
+            processHandler?.addProcessListener(object : ProcessAdapter() {
+
+                override fun onTextAvailable(event: ProcessEvent, outputType: Key<*>) {
+                    if (outputType === ProcessOutputTypes.STDOUT ||
+                        outputType === ProcessOutputTypes.STDERR
+                    ) {
+                        if (event.text.contains("[Y/n]:")) {
+                            insertAssertionNow = true
+
+                            ApplicationManager.getApplication().executeOnPooledThread {
+                                while (insertAssertionNow) {
+                                    processHandler.processInput?.write("Y\n".toByteArray(Charset.defaultCharset()))
+                                }
+                            }
+                        } else {
+                            insertAssertionNow = false
+                        }
+                        runInEdt {
+                            RunContentManager.getInstance(project).toFrontRunContent(executor, processHandler)
```

Comment:
seems like it'd be annoying to force this into focus every time there's text

Reply:
Switching to initialFocus on the window only, rather than when text available